addons/web/static/src/search/action_menus/action_menus.js

```diff
@@ -98,6 +93,38 @@ export class ActionMenus extends Component {
         });
     }
 
+    async getPrintItems() {
+        const printActions = this.props.items.print || [];
+        const asyncFilter = async (arr, predicate) => {
+            const results = await Promise.all(arr.map(predicate));
+            return arr.filter((_v, index) => results[index]);
+        };
+        const filteredPrintActions = await asyncFilter(printActions, async (printAction) => {
+            if (printAction.domain) {
+                const recMatch = await this.orm.search(
+                    this.props.resModel,
+                    [
+                        ["id", "in", this.props.getActiveIds()],
+                        ...new Domain(printAction.domain).toList(),
+                    ],
+                    { limit: 1 }
+                );
+                return recMatch.length > 0;
```

Comment:
It is very sad to add an extra RPC request for each action (with a domain) every time we access to the form view (or every time you select a record in the tree view). In the case of account.move (because of the override), it adds 2 extra RPCs each time (before it was 3 RPCs to open a form view of account.move, now it is 5). IMO, the current implementation is too much inefficient.

(At least it should be lazy when you click on the "actions" button, and only have one RPC for all actions if there's no alternative).

Reply:
It's even worse on the invoice form view:
before:
![image](https://github.com/odoo/odoo/assets/9432572/6f696196-1868-4553-b039-35ae091d2f0f)
after:
![image](https://github.com/odoo/odoo/assets/9432572/49b30c8f-f589-49d0-89c5-be55800ff999)